### PR TITLE
Fix problem with subscription on way points during resumption

### LINKED ISSUE #2460: SDL does not send SubscribeWayPoints request to HMI in case of resumption
### Bug Report
SDL does not send SubscribeWayPoints request to HMI in case of resumption

##### Reproduction Steps
1. In case of resumption the application has to send
SubscribeWayPoints request to HMI if it has to restore subscription

##### Expected Behavior
SDL send SubscribeWayPoints request to HMI in case of resumption

##### Observed Behavior
SDL does not send SubscribeWayPoints request to HMI in case of resumption

##### OS & Version Information
* OS/Version: 
* SDL Core Version: 
* Testing Against:

### PR DESCRIPTION
**Pull request by ychernysheva**
Fixes #2460 

This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
This pull request is intended to fix problem with subscription on way points during resumption, when SDL does not send SubscribeWayPoints request to HMI in case of resumption. This problem is caused by absence of functionality, which may ensure the sending corresponding command to HMI during resumption.

MessageHelper class is responsible for creation messages to HMI in similar situations, and already has two in fact duplicate methods with identical functionality (SendUnsubscribedWayPoints and SendStopAudioPathThru). There is a possibility to add another duplicate method like SendSubscribedWayPoints, but better solution may be implementation one shared method for creation such type of messages (with any function ID, depending of situation).

That's why in this pull request CreateMessageWithFunctionID method is added to MessageHelper class, and methods SendUnsubscribedWayPoints and SendStopAudioPathThru are replaced by CreateMessageWithFunctionID method in all places, where they were used. Also call of CreateMessageWithFunctionID is added to method AddWayPointsSubscription of ResumeCtrlImpl class to solve problem, mentioned in corresponding issue.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by ychernysheva** [comment]
@theresalech This PR is ready for review.
Thank you!

**Comment by theresalech** [comment]
Hi @ychernysheva, issue 2460 is not currently planned for the Core 6.2 release, but we will work to include if time allows.

File: src/components/application_manager/src/resumption/resume_ctrl_impl.cc [hunk 1]
```diff
@@ -914,6 +914,10 @@ void ResumeCtrlImpl::AddWayPointsSubscription(
         saved_app[strings::subscribed_for_way_points];
     if (true == subscribed_for_way_points_so.asBool()) {
       application_manager_.SubscribeAppForWayPoints(application);
+      auto message = MessageHelper::CreateMessageWithFunctionID(
+          application_manager_,
+          hmi_apis::FunctionID::Navigation_SubscribeWayPoints);
+      application_manager_.GetRPCService().ManageHMICommand(message);
```
Comment (jacobkeeler) [inline src/components/application_manager/src/resumption/resume_ctrl_impl.cc:920]:
```suggestion
      if (!application_manager_.IsAnyAppSubscribedForWayPoints())
          auto message = MessageHelper::CreateMessageWithFunctionID(
              application_manager_,
              hmi_apis::FunctionID::Navigation_SubscribeWayPoints);
          application_manager_.GetRPCService().ManageHMICommand(message);
      }
      application_manager_.SubscribeAppForWayPoints(application);
```
You should check if any apps are already subscribed before sending this message

File: src/components/application_manager/include/application_manager/message_helper.h [hunk 2]
```diff
@@ -617,20 +617,32 @@ class MessageHelper {
...
+  DEPRECATED static bool SendStopAudioPathThru(ApplicationManager& app_mngr);
 
   /**
+   * @deprecated Deprecated in favor of method CreateMessageWithFunctionID
    * @brief Sends UnsubscribeWayPoints request
    * @return true if UnsubscribedWayPoints is send otherwise false
    */
-  static bool SendUnsubscribedWayPoints(ApplicationManager& app_mngr);
+  DEPRECATED static bool SendUnsubscribedWayPoints(
+      ApplicationManager& app_mngr);
```
Comment (jacobkeeler) [inline src/components/application_manager/include/application_manager/message_helper.h:636]: These can be removed instead, due to the fact that the next release is a major release.

**Comment by ychernysheva** [comment]
@jacobkeeler Unfortunately, now this PR is outdated, because fix of this defect is a part of new feature, which will be delivered as part of release 7.0. Realization of affected functionality, responsible for processing subscription on way points, will be different in planned feature, that's way there is no need to deal with this fix now. Also all related changes, which are still actual, will be moved to mentioned feature.

May I suppose to close this PR as outdated?

**Comment by jacobkeeler** [comment]
@ychernysheva Which feature are you referring to?

**Comment by ychernysheva** [comment]
@jacobkeeler This feature is implementation of proposal [SDL-0190] Handle response from HMI during resumption data (issue #[2486](https://github.com/smartdevicelink/sdl_core/issues/2486))

**Comment by iCollin** [comment]
Implemented in #3475

**Pull request closed without merging**